packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -300,15 +302,19 @@ export class ProjectDataSource {
         return defaultFileName
       }
 
-      const specFileName = getDefaultSpecFileName(specPatternSet, this.ctx.lifecycleManager.fileExtensionToUse)
+      if (specPatternSet === defaultSpecPattern[this.ctx.coreData.currentTestingType]) {
+        return defaultFileName
+      }
+
+      const specFileName = getDefaultSpecFileName(specPatternSet, this.ctx.coreData.currentTestingType, this.ctx.lifecycleManager.fileExtensionToUse)
 
       if (!specFileName) {
         return defaultFileName
       }
 
       return specFileName
     } catch {
-      return defaultFileName
+      return getDefaultFileName('e2e')
```

Comment:
It looks like we could just try/catch the awaited specPatterns call and abort on line 302 in the case of a failure and not need separate error logic here.

If we do need to keep this, though, why not use currentTestingType instead of 'e2e'?

Reply:
I left the try/catch but updated the testing type - the reason is bc `getDefaultSpecFileName` is using different packages and is safer to catch it if there's any possible error while trying to create the fileName depending of the specPattern